File: presentation/src/main/java/com/okaapi/kotlindemo/controllers/PostsController.java
package com.okaapi.kotlindemo.controllers;

import com.okaapi.kotlindemo.model.PostDto;
import com.okaapi.kotlindemo.services.PostsService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.stream.Collectors;

@RestController
public class PostsController {

    private final PostsService postsService;

    public PostsController(PostsService postsService) {
        this.postsService = postsService;
    }

    @GetMapping("/posts")
    public List<PostDto> getPosts() {
        return postsService.getPosts()
                .stream()
                .map(post -> new PostDto(post.getTitle(), post.getContent()))
                .collect(Collectors.toList());
    }
}

File: presentation/src/main/kotlin/com/okaapi/kotlindemo/controllers/PostsController.kt
package com.okaapi.kotlindemo.controllers

import com.okaapi.kotlindemo.model.Post
import com.okaapi.kotlindemo.model.PostDto
import com.okaapi.kotlindemo.model.toDto
import com.okaapi.kotlindemo.services.PostsService
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Flux

@RestController
class PostsController(val postsService: PostsService) {

    @GetMapping("/posts")
    suspend fun getPosts() : Flow<PostDto> {
        return postsService.getPosts().map(Post::toDto)
    }

}